File: LeetCode/202HappyNumber/HappyNumber.cpp
class Solution {
public:
    bool isHappy(int n) {
        // Run Till Cycle -> Time:O(n), Space:O(n)
        set<int> seen_nums;
        int sum;
        while (seen_nums.find(n) == seen_nums.end()) {
            seen_nums.insert(n);
            sum = 0;
            while (n != 0) {
                sum += (n%10)*(n%10);
                n = n/10;
            }
            n = sum;
        }
        if (n == 1) {
            return true;
        }
        return false;
    }
};

File: LeetCode/202HappyNumber/HappyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        # Run Till Cycle -> Time:O(n), Space:O(n)
        seen_nums = set()
        while n not in seen_nums:
            seen_nums.add(n)
            num_sum = 0
            while n != 0:
                num_sum += (n%10)**2
                n = n//10
            n = num_sum
        if n == 1:
            return True
        return False
